api/classes/errors.js

```diff
@@ -1,31 +1,43 @@
 var Boom = require('boom');
-var logger = require('../../lib/logger');
 
 module.exports.attrs =  function(request, reply, source, error) {
   var failedAttribute = '`' + error.data.details[0].path + '`';
 
   // Look to see if the attribute was passed at all
   if (error.data.details[0].message.indexOf('is required') !== -1) {
-    return reply(Boom.badRequest(failedAttribute + ' must be passed.'));
+    return reply(Boom.badRequest(failedAttribute + ' must be passed.', {
+      debug: true,
+      error: failedAttribute + ' must be passed.'
+    }));
   }
 
   // Otherwise the type was invalid
-  return reply(Boom.badRequest(failedAttribute + ' invalid'));
+  return reply(Boom.badRequest(failedAttribute + ' invalid', {
+    debug: true,
+    error: failedAttribute + ' invalid'
+  }));
 };
 
 module.exports.id = function(request, reply, source, error) {
-  return reply(Boom.badRequest('`id` invalid'));
+  return reply(Boom.badRequest('`id` invalid', {
+    debug: true,
+    error: '`id` invalid'
+  }));
 };
 
 module.exports.name = function(request, reply, source, error) {
-  return reply(Boom.badRequest('`name` invalid'));
+  return reply(Boom.badRequest('`name` invalid', {
+    debug: true,
+    error: '`name` invalid'
+  }));
 };
 
 module.exports.generateErrorResponse = function(e) {
-  logger.error(e);
   if (e.isBoom) {
     return e;
```

Comment:
This whole approach isn't ideal since it isn't clear how these pieces fit together.

But. This function ensures that whatever error is thrown in a prerequisite or handler function is wrapped as a `boom` by the time it hits the extension function in `extensions.js`. That's where the actual logging happens.